### ISSUE TITLE
Improve Input Descriptions

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#2434]
Where inputs have associated descriptions that provide important information to the user, we should associate these descriptions programmatically with the input by using an aria-describedby attribute. 

- [x] inputs on the Contributors page (Name, Slug, Description) contain important information after the input. Add an aria-describedby attribute to the input and associate it with the ID of the details text so that assistive technology users hear these descriptions after the label 
![Screenshot from 2021-10-12 15-40-22](https://user-images.githubusercontent.com/13485451/137038540-329b77be-3be9-464f-9e75-28b38b62e7c5.png)
- [x] input description on the Edit Book information page. 
![Screenshot from 2021-10-12 15-40-32](https://user-images.githubusercontent.com/13485451/137038539-a64afbdd-71a3-44f7-ae6c-5aa0e07db939.png)
- [x] input descriptions on the Theme Options page 
![Screenshot from 2021-10-12 15-40-40](https://user-images.githubusercontent.com/13485451/137038537-ac041876-f07c-4bad-8b9a-25c85478bd28.png)

## Grouped inputs
When groups of checkboxes or radio buttons relate to the same question or setting each input has two labels, a label for the group and a label for the individual input. To associate each input with both typically a `<fieldset>` element is used to group related options, and the label for the group is contained in a <legend> element. There are several instances where inputs are not programmatically associated with their group labels.

- [ ] Radio buttons on the LTI settings page
![Screenshot from 2021-10-12 15-41-58](https://user-images.githubusercontent.com/13485451/137038644-57f94baa-56cd-4817-b0a5-b859c0184776.png)
- [ ] Radio buttons on the Theme Options page. In this case the radio buttons are grouped together with a <fieldset> element, but the label for the group is not within a `<legend>`
![Screenshot from 2021-10-12 15-41-48](https://user-images.githubusercontent.com/13485451/137038646-4032193f-c490-4eb6-a0fc-143393086318.png)

**Comment by SteelWagstaff** [comment]
Lisa recommends the following to help improve combobox inputs (like the Author input on the Book Edit page). 
* https://www.24a11y.com/2019/select-your-poison/
* https://www.24a11y.com/2019/select-your-poison-part-2/
Part 2 proposes a pattern that is very similar to what you currently have (the buttons above combobox input in the Multiselect section). Based on the user research they did it performed really well with users so I think it will make for a very accessible solution.

**Comment by fdalcin** [comment]
### Update

We have a draft PR that covers Book Info, Contributors and Theme Options pages. All the basic inputs fields should be working properly with `aria-describedby`.

For the custom combo boxes and custom text areas we might need to discuss how to approach.

Grouped inputs like the radio buttons should be fixable as long as we have control over the HTML. The LTI provider seems to be easy fixable since we have full control, see:
https://github.com/pressbooks/pressbooks-lti-provider-1p3/blob/dev/templates/book/settings.blade.php#L62
https://github.com/pressbooks/pressbooks-lti-provider-1p3/blob/dev/templates/network/settings.blade.php#L101

**Comment by SteelWagstaff** [comment]
@fdalcin for the fieldset on the global theme options page, I'd like to add the following legend: `<legend class="screen-reader-text">Choose a section level license option:</legend>`

W3C suggests that `Radio button groups should always be grouped using <fieldset>.` I'm going to work on a possible implementation in the LTI 1.3 plugin and share a draft PR when its ready.

**Comment by greatislander** [comment]
@fdalcin and @SteelWagstaff — I think rather than using JS for many of these, the pressbooks Options class could be modified to properly associate descriptions with the related fields. See, for example: https://github.com/pressbooks/pressbooks/blob/dev/inc/class-options.php#L243-L277

**Comment by greatislander** [comment]
For contributors (as well as chapter types, front matter types, and back matter types), this issue will be mostly resolved via this core change: https://core.trac.wordpress.org/ticket/55651 (arriving with WordPress 6.1).

**Comment by greatislander** [comment]
I think the Book Information page should be addressed with #2749.

**Comment by SteelWagstaff** [comment]
@greatislander will you please investigate our idea for systematically adding a screenreader only legend to fieldsets, as described here: https://github.com/pressbooks/pressbooks/issues/2434#issuecomment-964350124

**Comment by SteelWagstaff** [comment]
See https://youtu.be/N3DzGOvAxc4 for conversation with Ned about this issue and https://github.com/pressbooks/pressbooks/pull/2493

**Comment by SteelWagstaff** [comment]
Can potentially replace this issue with findings from latest audit. FYI @MichelleWeremczuk

**Comment by arzola** [comment]
This issue has been automatically closed as part of an organization-wide backlog review.

**Reason:** This issue was created 4y 4m ago with no updates in the last 1y 2m.

If this is still relevant, please feel free to reopen it with updated context. We appreciate your contributions and want to ensure our issue tracker stays actionable.

**Learn more:** https://pressbooks.community/t/announcement-automatic-closure-of-old-github-tickets/4542

**Comment by greatislander** [comment]
All remaining issues relevant to this ticket are tracked in pressbooks/private#1639.
